go/tasks/plugins/k8s/sagemaker/builtin_training.go

```diff
@@ -116,8 +116,14 @@ func (m awsSagemakerPlugin) buildResourceForTrainingJob(
 
 	inputModeString := strings.Title(strings.ToLower(sagemakerTrainingJob.GetAlgorithmSpecification().GetInputMode().String()))
 
-	role := awsUtils.GetRole(ctx, cfg.RoleAnnotationKey, taskCtx.TaskExecutionMetadata().GetAnnotations())
-	if role == "" {
+	role := awsUtils.GetRoleFromSecurityContext(taskCtx.TaskExecutionMetadata().GetSecurityContext())
+
+	// Continue this for backward compatibility
+	if len(role) == 0 {
```

Comment:
We should probably encapsulate the backward compat code in one place... so we can deprecate it easier and avoid new plugins having to know to do that...

Reply:
Cool.